### PR TITLE
@orta => Registration Flow Improvements

### LINKED ISSUE #436: new registration confirmation screen
image

image

### LINKED ISSUE #437: eliminate interstitial screen between finishing reg. and seeing details
image

Don't know when this got added but this screen should just lead to the users bidder details

### PR DESCRIPTION
**Pull request by ashfurrow**
@orta, this view controller looks like it _was_ doing a bunch at one point, but not anymore. So I removed things that didn't appear to be used (like the bid details view that was never being accessed). Please take a look over and see if everything looks good. 

Before:

screen shot 2015-04-30 at 10 03 36 pm

After:

screen shot 2015-04-30 at 10 25 01 pm

@katarinabatina I had to eyeball the line heights. Could you verify these are correct-ish? 

Fixes #436.
Fixes #437.

### CODE REVIEW

File: Kiosk/Bid Fulfillment/LoadingViewController.swift [hunk 1]
```diff
@@ -126,7 +126,7 @@ public class LoadingViewController: UIViewController {
         let showPlaceHigherButton = placingBid && (!isHighestBidder || reserveNotMet)
         placeHigherBidButton.hidden = !showPlaceHigherButton
 
-        let showAuctionButton = !placingBid || createdNewBidder
+        let showAuctionButton = showPlaceHigherButton || isHighestBidder || (!placingBid && !createdNewBidder)
```
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/LoadingViewController.swift]: This is quite possibly the hairiest part of the app. Another good candidate for MVVM.

File: Kiosk/Bid Fulfillment/LoadingViewController.swift [hunk 2]
```diff
@@ -137,6 +137,10 @@ public class LoadingViewController: UIViewController {
         titleLabel.text = "Registration Complete"
         bidConfirmationImageView.image = UIImage(named: "BidHighestBidder")
         fulfillmentContainer()?.cancelButton.setTitle("DONE", forState: .Normal)
+        RACSignal.interval(1, onScheduler: RACScheduler.mainThreadScheduler()).take(1).subscribeCompleted { [weak self] () -> Void in
```
Comment (orta) [inline Kiosk/Bid Fulfillment/LoadingViewController.swift]: I've seen this pattern a lot, maybe it should be turned into something with a name that actually represents what it means?

File: Kiosk/Storyboards/Fulfillment.storyboard [hunk 3]
```diff
@@ -802,65 +801,54 @@
...
+                                        <rect key="frame" x="65" y="77" width="182" height="43"/>
+                                        <color key="backgroundColor" white="0.0" alpha="0.0" colorSpace="calibratedWhite"/>
+                                        <fontDescription key="fontDescription" name="AGaramondPro-Semibold" family="Adobe Garamond Pro" pointSize="40"/>
+                                        <nil key="highlightedColor"/>
+                                    </label>
+                                    <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" fixedFrame="YES" text="An email and text message have been sent to you with these details. We encourage you to write the down as well." lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="WP8-Mg-tir" customClass="ARSerifLabel">
```
Comment (alloy) [inline Kiosk/Storyboards/Fulfillment.storyboard]: “write the down” ?

Comment (ashfurrow) [inline Kiosk/Storyboards/Fulfillment.storyboard]: Ha, good catch! Should be "them"

File: Kiosk/Bid Fulfillment/LoadingViewController.swift (hunk 2 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/LoadingViewController.swift]: The pattern of "wait X seconds"? We use that in two other places – syncing and pinging – but those are both recurring.

**Comment by ashfurrow** [comment]
The https://github.com/artsy/eidolon/commit/73b269f801bf8a70d5617292b7e5dee8adff58c3 commit is work towards fixing a crash I noticed in development. Shouldn't take too much longer, then I'll be on eigen.

File: Kiosk/Bid Fulfillment/LoadingViewController.swift (hunk 2 above)
Comment (orta) [inline Kiosk/Bid Fulfillment/LoadingViewController.swift]: ah, I thought this was a run on the runloop thing in a sec

**Comment by orta** [comment]
:+1:

File: Kiosk/Bid Fulfillment/LoadingViewController.swift (hunk 2 above)
Comment (ashfurrow) [inline Kiosk/Bid Fulfillment/LoadingViewController.swift]: Ah, nah it's just a fancy `dispatch_after` :joy:

**Pull request merged**